File: 03_A/C#/CSV_Reader/StandardDeviation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSV_Reader {
    class StandardDeviation {
        private OnlineMean mean;
        private double stdDev = 0;

        public double computeStdDev(List<double> data) {
            mean = new OnlineMean();
            foreach (double val in data) {
                mean.NewValue(val);
            }
            double resMean = mean.GetAvg();
            foreach (double val in data) {
                stdDev += Math.Pow((val - resMean), 2);
            }
            stdDev = Math.Sqrt(stdDev / data.Count);

            return stdDev;
        }

    }
}

File: 03_A/VB/CSV_Reader/StandardDeviation.vb
Public Class StandardDeviation
    Private mean As CSV_Reader.OnlineMean
    Private stdDev As Double = 0

    Public Function computeStdDev(ByVal data As List(Of Double)) As Double
        Dim mean As New OnlineMean()

        For Each value In data
            mean.NewValue(value)
        Next

        Dim resMean As Double = mean.GetAvg()

        For Each value In data
            stdDev += Math.Pow(value - resMean, 2)
        Next

        stdDev = Math.Sqrt(stdDev / data.Count)
        Return stdDev
    End Function
End Class
